### PR TITLE
feat(iroh): Improve documentation and canonicalize docs in `iroh::client`

### PR DESCRIPTION
**Pull request by matheus23**
## Description

- More pointers between individual modules to help people understand how they go from `A` to `B` (e.g. `Node` -> `Iroh`, `Iroh` -> `blobs::Client`, etc.)
- Documentation on how to initialize `client::Iroh`
- Generally make methods adhere to our documentation style.
- Expose the type alias `iroh::client::RpcClient`, so it doesn't show up as `RpcClient<RpcService, Connection<RpcService>>` everywhere it's used in the docs.

## Breaking Changes

None, but noteworthy:
- Expose the type alias `iroh::client::RpcClient`.

## Change checklist

- [X] Self-review.
- [X] Documentation updates following the [style guide](https://rust-lang.github.io/rfcs/1574-more-api-documentation-conventions.html#appendix-a-full-conventions-text), if relevant.
- ~~[ ] Tests if relevant.~~
- [X] All breaking changes documented.

### CODE REVIEW

File: iroh/src/client/docs.rs [hunk 1]
```diff
@@ -306,7 +312,7 @@ impl Doc {
         Ok(res.entry.map(|entry| entry.into()))
     }
 
-    /// Get entries.
+    /// Returns entries.
```
Comment (flub) [inline iroh/src/client/docs.rs:315]:
```suggestion
    /// Returns all entries matching the query.
```
?

File: iroh/src/client/docs.rs [hunk 2]
```diff
@@ -378,14 +384,14 @@ impl Doc {
...
-    /// Get status info for this document
+    /// Returns status info for this document
     pub async fn status(&self) -> anyhow::Result<OpenState> {
         self.ensure_open()?;
         let res = self.rpc(StatusRequest { doc_id: self.id() }).await??;
         Ok(res.status)
     }
 
-    /// Set the download policy for this document
+    /// Sets the download policy for this document
```
Comment (flub) [inline iroh/src/client/docs.rs:394]: Things like this should probably explain a bit what download policies are and what is available and how they affect things.

But I'll take any improvement, so like this is also fine.

**Review by flub — APPROVED** [review]
**adding** a public api is not a breaking change is it?

File: iroh/src/client/docs.rs (hunk 2 above)
Comment (matheus23) [reply iroh/src/client/docs.rs:394]: Agreed. Happy to pick this up on another docs day :)

**Pull request merged**